features/poller/api/src/main/java/org/opennms/netmgt/poller/MonitoredServiceTask.java

```diff
@@ -69,5 +100,35 @@ public void setMonitoredService(MonitoredService m_monitoredService) {
 	public void setParameters(Map<String, Object> parameters) {
 		this.m_parameters = parameters;
 	}
+	
+	/**
+	 * 
+	 * @param location
+	 */
+	public void setLocation(String location){
+		m_location = location;
+	}
+	
+	/**
+	 * 
+	 * @return location
+	 */
+	public String getLocation(){
+		 return m_location == null || "".equals(m_location) ? "localhost" : m_location;
+	}
+	
 
+    /**
+     * <P>
+     * Returns the total task timeout in milliseconds for all IP ranges.
+     * </P>
+     */
+    public int calculateTaskTimeout() {
+        BigDecimal taskTimeOut = BigDecimal.ZERO;
+        BigDecimal timeout =  m_parameters.get("timeout") == null ? BigDecimal.valueOf(1) : (BigDecimal)m_parameters.get("timeout");
```

Comment:
I thought about it more and I think that if either the "timeout" or "retry" parameters are missing, the default timeout should be equal to the polling interval (usually 5 minutes). Try and fetch this value from the configuration and use it as the JMS timeout.